### ISSUE TITLE
Every check of `isStandardSyntaxSelector` but one reads a quoted attribute value as selector syntax

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#151]
Found while reviewing the fix for #125, which taught one check of `isStandardSyntaxSelector` to read past a quoted attribute value. The others still do not, and each of them turns a stretch of ordinary CSS into something the whole plugin passes over.

### What minimal example or steps are needed to reproduce the bug?

```css
[title=":extend(x)"] { color: red; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/string-quotes": "single"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`Expected single quotes`, and `--fix` writing `[title=':extend(x)']`.

### What actually happened?

Nothing is reported, and nothing is fixed. `isStandardSyntaxSelector` reads the text inside the quotes as selector syntax, decides the selector is a Less `:extend`, and every rule built on the util passes the rule node over.

The same holds for each of the other checks that match on text alone. All of these are ordinary CSS, and all are declared non-standard:

```css
[title=":extend(x)"] {}
[title="//"] {}
[title="<%"] {}
[title="%>"] {}
[title=".foo()bar"] {}
```

An attribute value may hold any text at all, so any of them can turn up in a real stylesheet — a selector matching a link by its `href`, or a `[data-*]` carrying a template fragment.

The rules that fall silent are the ones the util speaks for: `no-extra-semicolons`, `selector-attribute-brackets-space-inside`, `selector-descendant-combinator-no-non-space`, `selector-list-comma-newline-after`, `selector-pseudo-class-case`, `selector-pseudo-class-parentheses-space-inside`, `selector-pseudo-element-case` and `string-quotes`.

### Do you have a proposal to fix the bug?

The util already has `withoutQuotedText`, added in #125 for the Less guard check. Every check that matches on text rather than on structure wants the same treatment: the interpolation, placeholder, nested-property, `:extend`, mixin, ERB and comment tests all read the raw selector today.

Two things are worth settling before moving them over.

The mixin tests are not purely textual — `/\.[\w-]+\(.*\).+/u` and `selector.endsWith(')')` describe a shape, and blanking the text inside quotes leaves the quotes and the length alone, so the shape survives. They should be safe, but each wants a case of its own to say so.

`hasInterpolation` is a util in its own right and is used elsewhere, so `#{…}` inside a quoted value is a question about that util rather than this one, and probably belongs in a separate change.

The behaviour is as old as the checks themselves, and #125 changed none of it — it only made the same blindness visible in one place, and cured it there.

**Comment by firefoxic** [comment]
The changelog entries that came of this issue and of the ones after it — [#164](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/164), [#170](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/170) and [#171](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/171) — say "several rules" and point here rather than carrying a list of names each. This is the list they point to, and it is fifteen rules rather than the eight this issue names above.

A rule is on it if it asks whether the selector it is about is a selector at all before it reads anything, so a wrong answer to that question silences it. Eight ask the question themselves:

- `no-extra-semicolons`
- `selector-attribute-brackets-space-inside`
- `selector-descendant-combinator-no-non-space`
- `selector-list-comma-newline-after`
- `selector-pseudo-class-case`
- `selector-pseudo-class-parentheses-space-inside`
- `selector-pseudo-element-case`
- `string-quotes`

Seven more ask it through a checker they share, which is why they were missed:

- `selector-list-comma-newline-before`, `selector-list-comma-space-before` and `selector-list-comma-space-after`
- `selector-combinator-space-before` and `selector-combinator-space-after`
- `selector-attribute-operator-space-before` and `selector-attribute-operator-space-after`

Each of the seven was confirmed by giving it a selector it fixes and the same selector carrying a Less `:extend`, which is a preprocessor construct the plugin passes over on purpose. `[title = "x"] {}` comes back as `[title= "x"] {}` under `selector-attribute-operator-space-before: never`, while `[title = "x"]:extend(.z) {}` is left exactly as it stands; and so on for the other six.
